synthtool/languages/node_mono_repo.py

```diff
@@ -57,11 +57,15 @@ def read_metadata(relative_dir: str):
         )
 
         repo = git.parse_repo_url(repo_url)
-
+        data["directory_path"] = (
+            data["repository"]
+            if isinstance(data["repository"], str)
+            else f'{data["repository"]["directory"]}'
```

Comment:
The existence of the `repository.directory` field was a later addition, so it might throw an error if it doesn't exist. Perhaps better to add another fallback.